### PR TITLE
2.28 jars + process XML

### PR DESCRIPTION
**Pull request by wpride**
This started off as a fix to http://manage.dimagi.com/default.asp?225962

In order to get an error message for missing case_id (or other invalid XML structures) we need to process the instance XML in Touchforms. To do this without the SQLite backend we can use the MockUserDataSandbox, so this package is added to the API library.

I think this makes sense to QA along with the 2.28 Jars so I added those as well, ran into two other issues. 

https://github.com/dimagi/commcare/commit/9fd15c930fe2b331fb3194b1528f5edcb6cfee12
https://github.com/dimagi/commcare/commit/bd32d33c2ff6b2a951775c1981d3d91c46d434fd

(apologies Phillip) 

The first moved the getSessionInstance function into a static helper, the second added another 1-arg CommCareSession constructor and Jython doesn't give us a way to specify one or the other so I needed to add a default constructor  - see https://github.com/dimagi/commcare/pull/298/files

@benrudolph

### CODE REVIEW

File: touchforms/backend/touchcare.py [hunk 1]
```diff
@@ -168,8 +168,9 @@ def from_bundle(inst):
             for k, v in self.vars.get('user_data', {}).iteritems():
                 clean_user_data[k] = unicode(v if v is not None else '', errors='replace')
 
-            return from_bundle(sess.getSessionInstance(*([self.vars.get(k, '') for k in meta_keys] +
-                                                         [to_hashtable(clean_user_data)])))
+            form_instance = SessionInstanceBuilder.getSessionInstance(sess.getFrame(), *([self.vars.get(k, '') for k in meta_keys] +
```
Comment (dimagimon) [inline touchforms/backend/touchcare.py]: E501 line too long (132 > 115 characters)

File: touchforms/backend/touchcare.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def from_bundle(inst):
         elif 'session' in ref:
             meta_keys = ['device_id', 'app_version', 'username', 'user_id']
             exclude_keys = ['additional_filters', 'user_data']
-            sess = CommCareSession(None)  # will not passing a CCPlatform cause problems later?
+            sess = CommCareSession()  # will not passing a CCPlatform cause problems later?
```
Comment (benrudolph) [inline touchforms/backend/touchcare.py]: what is this change for?

File: touchforms/backend/touchcare.py [hunk 3]
```diff
@@ -198,9 +199,12 @@ def process_form_file(auth, submission_file, session_data=None):
...
 
-def process_form_xml(auth, submission_xml, session_data=None):
-    ccInstances = CCInstances(session_data, auth, uses_sqlite=True)
-    sandbox = ccInstances.sandbox
+def process_form_xml(auth, submission_xml, uses_sqlite, session_data=None):
+    if uses_sqlite:
+        ccInstances = CCInstances(session_data, auth, uses_sqlite=True)
+        sandbox = ccInstances.sandbox
+    else:
+        sandbox = MockDataUtils.getStaticStorage()
```
Comment (benrudolph) [inline touchforms/backend/touchcare.py]: maybe a comment here as to why we need to mock this?

File: touchforms/backend/touchcare.py (hunk 2 above)
Comment (benrudolph) [inline touchforms/backend/touchcare.py]: nvm saw other PR

File: touchforms/backend/touchcare.py [hunk 4]
```diff
@@ -26,6 +25,7 @@
 from org.commcare.core.sandbox import SandboxUtils
 from org.commcare.modern.process import FormRecordProcessorHelper as FormRecordProcessor
 from org.commcare.modern.parse import ParseUtilsHelper as ParseUtils
+from org.commcare.util.mocks import MockDataUtils
```
Comment (dimagimon) [inline touchforms/backend/touchcare.py]: F401 'MockDataUtils' imported but unused

**Comment by wpride** [comment]
passed QA, merging

File: touchforms/backend/xformplayer.py [hunk 5]
```diff
@@ -726,15 +726,13 @@ def submit_form(xform_session, answers, prevalidated):
     else:
         resp = form_completion(xform_session)
         resp['status'] = 'success'
-        xml = xform_session.output()
-        if xform_session.uses_sql_backend:
```
Comment (benrudolph) [inline touchforms/backend/xformplayer.py]: not everyone is on sqlite backend, will this break that?

**Pull request merged**